forte/processors/data_augment/algorithms/skeleton_op.py

```diff
@@ -0,0 +1,765 @@
+            children = [entry.get_parent(), entry.get_child()]  # type: ignore
+        else:
+            children = entry.get_members()  # type: ignore
+
+        # Copy the children entries.
+        new_children: List[Entry] = []
+        for child_entry in children:
+            if isinstance(child_entry, (Link, Group)):
+                # Recursively copy the children Links/Groups.
+                if not self._copy_link_or_group(
+                    child_entry, entry_map, new_pack
+                ):
+                    return False
+            else:
+                # Children Annotation must have been copied.
+                if child_entry.tid not in entry_map:
+                    return False
+            new_child: Entry = new_pack.get_entry(entry_map[child_entry.tid])
+            new_children.append(new_child)
+
+        # Create the new entry and add to the new pack.
+        new_entry: Entry
+        if is_link:
+            entry = cast(Link, entry)
+            new_link_parent: Entry = new_children[0]
+            new_link_child: Entry = new_children[1]
+            new_entry = type(entry)(
+                new_pack, new_link_parent, new_link_child  # type: ignore
+            )
+        else:
+            entry = cast(Group, entry)
+            new_entry = type(entry)(new_pack, new_children)  # type: ignore
+        new_pack.add_entry(new_entry)
+        entry_map[entry.tid] = new_entry.tid
+        return True
+
+    def _copy_multi_pack_link_or_group(
+        self, entry: Union[MultiPackLink, MultiPackGroup], multi_pack: MultiPack
+    ) -> bool:
+        r"""
+        This function copies a MultiPackLink/MultiPackGroup in the multipack.
+        It could be used in tasks such as text generation, where
+        MultiPackLink is used to align the source and target.
+        Args:
+            entry: The MultiPackLink/MultiPackGroup to copy.
+            multi_pack: The multi_pack contains the input entry.
+        Returns:
+            A bool value indicating whether the copy happens.
+        """
+        # The entry should be either MultiPackLink or MultiPackGroup.
+        is_link: bool = isinstance(entry, BaseLink)
+        children: List[Entry]
+        if is_link:
+            children = [entry.get_parent(), entry.get_child()]  # type: ignore
+        else:
+            children = entry.get_members()  # type: ignore
+
+        # Get the copied children entries.
+        new_children: List[Entry] = []
+        for child_entry in children:
+            child_pack: DataPack = child_entry.pack
+            child_pack_pid: int = child_pack.pack_id
+            # The new pack should be present.
+            if (
+                child_pack_pid not in self._data_pack_map
+                or child_pack_pid not in self._entry_maps
+            ):
+                return False
+            new_child_pack: DataPack = multi_pack.get_pack_at(
+                multi_pack.get_pack_index(self._data_pack_map[child_pack_pid])
```

Comment:
I feels like this can be done via this func: 
https://github.com/asyml/forte/blob/1fac83be0b2d34d8adb995a2629e31743756e71b/forte/data/multi_pack.py#L275

Reply:
It is using the same function